src/Orchard.Web/Modules/Orchard.Setup/Services/WebconfigOverrides.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Configuration;
+
+namespace Orchard.Setup.Services
+{
+
+    public class WebconfigOverrides
+    {
+        private const string AppSettingDefaultDataProvider = "defaultProvider";
+        private const string AppSettingDefaultRecipe = "defaultRecipe";
+        private const string ConnectionStringDefaultConnectionString = "defaultConnection";
+
+        public string DataProvider { get; set; }
+        public string DataConnectionString { get; set; }
+        public string DefaultRecipe { get; set; }
+
+        public static WebconfigOverrides Load()
+        {
+            var result = new WebconfigOverrides();
+
+            if (ConfigurationManager.ConnectionStrings[ConnectionStringDefaultConnectionString] != null)
```

Comment:
Yes, when running in an Azure App Service environment, the configuration settings specified in the Portal (through ARM) are available as environment variables at runtime. For each app setting, two environment variables are created; one with the name specified by the app setting entry, and another with a prefix of APPSETTING_. Both contain the same value.

For .NET apps in particular, these settings are also injected into the .NET configuration AppSettings/ConnectionString at runtime, overriding existing settings. That's why we can transparently use the ConfigurationManager class to read the settings.